src/blocks/available.ts

```diff
@@ -45,31 +49,91 @@ export function testMatchPatterns(
   return re.test(url);
 }
 
+function testUrlPattern(
+  pattern: string | URLPatternInit,
+  url: string = document.location.href
+): boolean {
+  let compiled;
+
+  try {
+    compiled = new URLPattern(pattern);
+  } catch {
+    if (typeof pattern === "object") {
+      for (const [key, entry] of Object.entries(pattern) as Entries<
+        typeof pattern
+      >) {
```

Comment:
Actually we could use this package across the repo (companion to `type-test`) that does this automatically, without much code at all: https://github.com/sindresorhus/ts-extras/blob/main/source/object-entries.ts
```ts
import { objectEntries } from "ts-extras";
```

```suggestion
      for (const [key, entry] of objectEntries(pattern)) {
```

Reply:
I'd be in favor of using that in the future. Object.entries definitely one of the rough edges also when using nominal types for records